recipes/jenkins.rb

```diff
@@ -20,39 +20,34 @@
 # limitations under the License.
 #
 
-# These plugins allow us to pull code (for jobs) from git
-jenkins_plugin "scm-api"
-jenkins_plugin "git-client"
-jenkins_plugin "git"
+include_recipe 'jenkins::master'
 
-# This colourises console output, instead of displaying literal ansi
-# escape sequences in the browser.
-jenkins_plugin "ansicolor"
+# unique plugins array before processing it
+node.set['jenkins_ci']['jenkins']['plugins'] = node['jenkins_ci']['jenkins']['plugins'].uniq.sort
+node['jenkins_ci']['jenkins']['plugins'].each do |plugin|
+  Chef::Log.info("---------------------------------> Installing plugin #{plugin} <---------------------------------")
+  jenkins_plugin plugin do
+    install_deps true
+    notifies :restart, 'service[jenkins]', :delayed
+  end
+end
 
-# This plugin lets us define static files which can (optionally) be
-# installed into a job's workspace before running any commands.  We
-# use it to override the test-kitchen configuration to use docker
-# instead of vagrant.
-cookbook_file "#{node['jenkins']['master']['home']}/custom-config-files.xml" do
+cookbook_file "#{node['jenkins']['master']['home']}/hudson.plugins.warnings.WarningsPublisher.xml" do
   owner "jenkins"
   group "jenkins"
   mode "0644"
-  notifies :restart, "service[jenkins]"
-end
-jenkins_plugin "token-macro"
-jenkins_plugin "config-file-provider" do
-  version "2.7"
-  action :install
+  notifies :restart, "service[jenkins]", :delayed
+  only_if { node['jenkins_ci']['jenkins']['warnings_publisher'] == true }
 end
 
-# This plugin lets us parse console output to report on warnings.
-# We'll use this to extract foodcritic's complaints, per the
-# instructions on http://acrmp.github.io/foodcritic/#ci
-jenkins_plugin "analysis-core"
-jenkins_plugin "warnings"
-cookbook_file "#{node['jenkins']['master']['home']}/hudson.plugins.warnings.WarningsPublisher.xml" do
+# This plugin lets us define static files which can (optionally) be
+# installed into a job's workspace before running any commands.  We
+# use it to override the test-kitchen configuration to use docker
+# instead of vagrant.
+cookbook_file "#{node['jenkins']['master']['home']}/custom-config-files.xml" do
   owner "jenkins"
   group "jenkins"
   mode "0644"
-  notifies :restart, "service[jenkins]"
+  notifies :restart, "service[jenkins]", :delayed
+  only_if { node['jenkins_ci']['jenkins']['custome_kitchen'] == true }
```

Comment:
I was going to ask if this was a typo ('custome') but I do see it set elsewhere.